### PR TITLE
Play well with others...

### PR DESCRIPTION
**Pull request by hitherto**
By accepting (and passing through) *args and **kwargs from our __init__() function, this library will work better in multi-inheritance scenarios (elb_instances.py monitoring script is the thing that needs this right now, but the change shouldn't impact functionality in other scenarios).

### CODE REVIEW

**Comment by plathrop** [comment]
I don't understand why this is necessary. As a subclass, in your `__init__`:
```python
def __init__(self, name, stuff, more_stuff=True):
    super(self, MyApplication).__init__(name)
```
makes this totally unnecessary?

File: krux_boto/cli.py [hunk 1]
```diff
@@ -25,9 +25,9 @@
 
 class Application(krux.cli.Application):
 
-    def __init__(self, name=NAME):
+    def __init__(self, name=NAME, *args, **kwargs):
         # Call to the superclass to bootstrap.
-        super(Application, self).__init__(name=name)
+        super(Application, self).__init__(name=name, *args, **kwargs)
```
Comment (hanpeter) [inline krux_boto/cli.py]: We can pass unlimited number of arguments to `krux.cli.Application` or `krux_boto.cli.Application` this way. However, these classes are not doing anything with it. So, is there any point in doing this? Can't you just overload the `__init__()` in your subclass and call the existing constructor for the superclass?

**Comment by hitherto** [comment]
@plathrop Because Python?

If we inherit like this (which we do in the elb monitor)...

`class Application(krux_monitoring_scripts.cli.Application, krux_boto.Application):`

...then when we call `super(self, Application).__init__([...])`, the call stack is actually...

`krux_monitoring_scripts.cli.Application.__init__([...])` -> `krux_boto.Application.cli.__init__([...])` -> `krux.cli.__init__([...])`

...in other words, because `krux_boto` sits in the middle of the stack, if it doesn't pass through any extra keywords, they don't arrive at the (shared) Super class (ie `krux.cli.Application`) above it. And if `krux_boto` isn't set up to receive extra args at all, we get a fatal error.

This can be alleviated if we switch the order of inheritance (ie: `class Application(krux_boto.Application, krux_monitoring_scripts.cli.Application):`) 

...except that then krux_boto's `run()` method takes precedence, which means the script doesn't do what we want it to...

Inheritance! It's fun! Also, terrible!

(obviously, the other solution would be to rewrite the elb monitor to use krux_boto without actually inheriting from the class alongside `krux_monitoring_scripts.cli`, but given the utility nature of this library, the ability to play well with multi-inheritance like this seems like a desirable feature anyhow?)

File: krux_boto/cli.py (hunk 1 above)
Comment (hitherto) [inline krux_boto/cli.py]: See note below.

**Comment by plathrop** [comment]
-1. Multiple inheritance is an abomination. We should do the work to make this work without multiple inheritance.

**Comment by hitherto** [comment]
Yeah, multiple inheritance is... bleh. One of the major reasons I still reflexively eye all OO patterns with a tiny bit of suspicion... :smile: 

I'm killing this for now. But might resurrect it if the alternatives turn out to all be even more horrible...

**Pull request closed without merging**